src/main/java/stream/flarebot/flarebot/commands/moderation/PurgeCommand.java

```diff
@@ -101,6 +103,80 @@ public void run() {
+                    MessageUtils.sendErrorMessage("That target user cannot be found, try mentioning them, using the user ID or using `all` to clear the entire chat.", channel);
+                    return;
+                }                
+            }
+
+            int amount = GeneralUtils.getInt(args[1], -1);
+
+            // 2 messages min
+            if(amount < 1) {
+                MessageUtils.sendErrorMessage("You must purge at least 1 message, please give a vaid purge amount.", channel);
+                return;
+            }
+
+            // This will be a successful delete so limit here.
+            if (!guild.getPermissions().isCreator(sender)) {
+                long riotPolice = cooldowns.computeIfAbsent(channel.getGuild().getId(), n -> 0L);
+                if (System.currentTimeMillis() - riotPolice < cooldown) {
+                    channel.sendMessage(MessageUtils.getEmbed(sender)
+                            .setDescription(String
+                                    .format("You are on a cooldown! %s seconds left!",
+                                            (cooldown - (System
+                                                    .currentTimeMillis() - riotPolice)) / 1000))
+                            .build()).queue();
+                    return;
+                }
+            }
+
+            if(!guild.getGuild().getSelfMember().hasPermission(Permission.MESSAGE_MANAGE, Permission.MESSAGE_HISTORY)) {
+                MessageUtils.sendErrorMessage("I do not have the perms `Manage Messages` and `Read Message History` please make sure I have these and do the command again.", channel);
+                return;
+            }
+            MessageHistory history = new MessageHistory(channel);
+            int toRetrieve = amount + 1;
+            int i = 0;
+            outer:
+            while(toRetrieve > 0) { // I don't really know if this should be min... 
+                                    // since deleting 10 of someone could be like 100 back yet this would request it 10 times. 
+                                    // For now I will just request 100 here each time.
+                if(history.retrievePast(100).complete().isEmpty()) {
+                    break;
+                }
+
+                List<Message> toDelete = new ArrayList<>();
+                for(Message msg : history.getRetrievedHistory()) {
+                    if(msg.getCreationTime().plusWeeks(2).isBefore(OffsetDateTime.now())) break outer;
+                    if(targetUser != null && msg.getAuthor().getId().equals(targetUser.getId()))
+                        toDelete.add(msg);
+                    else
+                        toDelete.add(msg);
+                    i++;
+                }
+                channel.deleteMessages(toDelete).complete();
+                toRetrieve -= toDelete.size();
+                toDelete.clear();
+            }
+            channel.sendMessage(MessageUtils.getEmbed(sender)
```

Comment:
We have a `MessageUtils#sendAutoDeletedMessage(MessageEmbed messageEmbed, long delay, MessageChannel channel)` for this

Reply:
Yep! New helper methods are always good good.